codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -277,9 +266,9 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
         public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int streamDependency,
                 short weight, boolean exclusive, int padding, boolean endOfStream) throws Http2Exception {
             Http2Stream stream = connection.stream(streamId);
-            verifyGoAwayNotReceived();
-            if (shouldIgnoreFrame(stream, false)) {
-                // Ignore this frame.
+
+            boolean ignoreFrame = streamCreatedAfterGoAwaySent(stream) || resetSentForStream(stream);
```

Comment:
Do we need to have this in a variable? can we just evaluate this directly in the `if (...)` conditional?